File: 2025_03/leetcode2025-03-11.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    int sumOfBeauties(vector<int>& nums) {
        int n = nums.size();
        vector<int> mn(n+1, INT_MAX), mx(n+1, 0);
        for (int i=1; i<=n; i++){
            mx[i] = max(mx[i-1], nums[i-1]);
            mn[n-i] = min(mn[n-i+1], nums[n-i]);
        }

        int res = 0;
        for (int i=1; i<n-1; i++){
            if (nums[i] > mx[i] && nums[i] < mn[i+1]){
                res += 2;
            }
            else if (nums[i] > nums[i-1] && nums[i] < nums[i+1]){
                res += 1;
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,3};
    cout << s.sumOfBeauties(nums) << endl;
}

File: leetcode2025/2025_03/leetcode2025-03-11.py
from typing import List

class Solution:
    def sumOfBeauties(self, nums: List[int]) -> int:
        myMax = lambda x,y:x if x>y else y
        myMin = lambda x,y:x if x<y else y
        n = len(nums)
        mx = [0]*(n+1)
        mn = [float("inf")]*(n+1)
        for i in range(1,n+1):
            mx[i] = myMax(mx[i-1], nums[i-1])
            mn[n-i] = myMin(mn[n-i+1], nums[n-i])
        
        res = 0
        for i in range(1,n-1):
            if nums[i] > mx[i] and nums[i] < mn[i+1]:
                res += 2
            elif nums[i] > nums[i-1] and nums[i] < nums[i+1]:
                res += 1
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,3]
    print(s.sumOfBeauties(nums))
